### ISSUE TITLE
espDevice.provision reports an error, session nil

### ISSUE THREAD

**Issue by mxdios** [espressif/esp-idf-provisioning-ios#21]
Excuse me

When SoftAP network configuration, call "provision" to report an error, session == nil, program crash

```
espDevice.provision(ssid: ssid!, passPhrase: passPhrase!, completionHandler: { (status) in
        print(status)
})
```

```
 public func provision(ssid: String, passPhrase: String = "", completionHandler: @escaping (ESPProvisionStatus) -> Void) {
        ESPLog.log("Provisioning started.. with ssid:\(ssid) and password:\(passPhrase)")
        if session == nil, !session.isEstablished { // session == nil , program crash
            completionHandler(.failure(.sessionError))
        } else {
            provisionDevice(ssid: ssid, passPhrase: passPhrase, retryOnce: true, completionHandler: completionHandler)
        }
    }
```

**Comment by vikas-chandra-mnnit** [comment]
@mxdios Thanks for reaching out to us, we will fix this issue in our next update and will notify you once its done.

**Comment by vikas-chandra-mnnit** [comment]
@mxdios We have fixed the crash in our library update 2.0.10. You may already know this but please make sure to call the `provision` API after `connect` API is successful.

**Comment by shahpiyushv** [comment]
Closing this issue since appropriate solution has been provided. Please fell free to reopen/file new issues if required.
